### ISSUE TITLE
[BUG] Race condition on resolve for `createDB()`

### ISSUE THREAD

**Issue by Sebastian-Webster** [Sebastian-Webster/mysql-memory-server-nodejs#42]
**This issue is in the latest version of mysql-memory-server**
Please make sure you have installed the latest version of `mysql-memory-server` and that this is still an issue in the latest version.
- [x]  This issue occurs in the latest version

**This issue has not been reported yet**
Please make sure this is not a duplicate of any other issue. You can [search here](https://github.com/Sebastian-Webster/mysql-memory-server-nodejs/issues) for existing issues
- [x] This issue has not been reported yet

**Describe the bug**
There is a rare race condition with `createDB()` where sometimes the function will resolve before the database has actually been started. This causes CI tests to sometimes fail at random because the test tries to connect to the database before it has actually been started

**To Reproduce**
It is a rare condition to reproduce, but this CI workflow run shows this behaviour https://github.com/Sebastian-Webster/mysql-memory-server-nodejs/actions/runs/10297357222/job/28500333285

**Expected behavior**
The function should not resolve until the database has started

**Actual behavior**
Most of the time the expected behaviour does actually happen, but there are a few rare times where the function will resolve before the database starts.

**Environment:**
 - OS and OS version: macOS 13.6.8
 - Node.js version: 19.9.0
 - mysql-memory-server version: 1.1.0

**Additional context**
It seems like this is happening because the current implementation is reading the log file and finding when it includes `started with:`. Just because it has started doesn't mean it has started listening for connections yet, which causes the race condition. This should be changed so it finds `ready for connections` from stderr. That would make the function resolve once the database is ready for connections instead of when the database has started.
